src/poprox_recommender/recommenders/configurations/nrms_topics_fb_filter.py

```diff
@@ -0,0 +1,210 @@
+        interacted_articles=e_clicked,
+        interest_profile=i_profile,
+    )
+
+    # Embed the user negative feedback
+    ue_neg_fb_config = UserArticleFeedbackConfig(
+        model_path=model_file_path("nrms-mind/user_encoder.safetensors"),
+        device=device,
+        feedback_type=False,
+    )
+    e_feedback_negative = builder.add_component(
+        "user-pos-fb-embedder",
+        UserArticleFeedbackEmbedder,
+        ue_neg_fb_config,
+        candidate_articles=e_candidates,
+        interacted_articles=e_clicked,
+        interest_profile=i_profile,
+    )
+
+    # Score articles based on interaction history
+    n_scorer = builder.add_component("scorer", ArticleScorer, candidate_articles=e_candidates, interest_profile=e_user)
+
+    # Score articles based on topic preferences
+    positive_topic_score = builder.add_component(
+        "positive_topic_score",
+        ArticleScorer,
+        candidate_articles=builder.node("candidate-embedder"),
+        interest_profile=e_topic_positive,
+    )
+
+    negative_topic_score = builder.add_component(
+        "negative_topic_score",
+        ArticleScorer,
+        candidate_articles=builder.node("candidate-embedder"),
+        interest_profile=e_topic_negative,
+    )
+
+    topic_fusion = builder.add_component(
+        "topic_fusion",
+        ScoreFusion,
+        {"combiner": "sub"},
+        candidates1=positive_topic_score,
+        candidates2=negative_topic_score,
+    )
+
+    # Score articles based on feedback
+    positive_feedback_score = builder.add_component(
+        "positive_feedback_score",
+        ArticleScorer,
+        candidate_articles=builder.node("candidate-embedder"),
+        interest_profile=e_feedback_positive,
+    )
+
+    negative_feedback_score = builder.add_component(
+        "negative_feedback_score",
+        ArticleScorer,
+        candidate_articles=builder.node("candidate-embedder"),
+        interest_profile=e_feedback_negative,
+    )
+
+    feedback_fusion = builder.add_component(
+        "feedback_fusion",
+        ScoreFusion,
+        {"combiner": "sub"},
+        candidates1=positive_feedback_score,
+        candidates2=negative_feedback_score,
+    )
+
+    # Combine topic scoring and feedback -> all explicit data
+    explicit_fusion = builder.add_component(
+        "explicit_fusion",
+        ScoreFusion,
+        {"combiner": "avg"},
+        candidates1=topic_fusion,
+        candidates2=feedback_fusion,
+    )
+
+    # Combine click and explicit feedback -> all preference
+    fusion = builder.add_component(
+        "fusion",
+        ScoreFusion,
+        {"combiner": "avg", "weight1": 1, "weight2": 2},
+        candidates1=n_scorer,
+        candidates2=explicit_fusion,
+    )
+
+    # Filter articles based on topic preferences
+    f_candidates = builder.add_component(
+        "topic-filter", TopicFilter, candidates=fusion, interest_profile=i_profile
+    )
+
+    r_filtered = builder.add_component(
```

Comment:
I _think_ this is the same as the one below it. Should the candidates be `f_candidates`?